File: ProjetAngular/src/app/pagesweb/model/inventaire.ts
import { ObjetInventaire } from './objet-inventaire';
import { Partie } from './partie';
export class Inventaire {
  public get partie(): Partie|undefined {
    return this._partie;
  }
  public set partie(value: Partie|undefined ) {
    this._partie = value;
  }
  public get objets(): ObjetInventaire[]|undefined {
    return this._objets;
  }
  public set objets(value: ObjetInventaire[]|undefined) {
    this._objets = value;
  }

  public get nom(): string|undefined {
    return this._nom;
  }
  public set nom(value: string |undefined) {
    this._nom = value;
  }
  public get id(): number|undefined {
    return this._id;
  }
  public set id(value: number|undefined) {
    this._id = value;
  }


  constructor(private _id?: number,
              private _nom?: string,
              private _objets?: ObjetInventaire[],
              private _partie?: Partie
              ){}
}

File: projetjeuSpringBoot/projetjeu/src/main/java/fr/projetjeu/model/Inventaire.java
package fr.projetjeu.model;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "inventaire")
public class Inventaire {


	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "inv_id")
	@JsonView(JsonViews.Common.class)
	private Integer id;
	
	
	@JsonView(JsonViews.InventaireAvecObjets.class)
	@OneToMany(mappedBy = "inventaire")
	private List<ObjetInventaire> objets;

	
	//@JsonView(JsonViews.Common.class)
	@OneToOne(mappedBy = "inventaire")
	private Partie partie;

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}


	public Partie getPartie() {
		return partie;
	}

	public void setPartie(Partie partie) {
		this.partie = partie;
	}

	public List<ObjetInventaire> getObjets() {
		return objets;
	}

	public void setObjets(List<ObjetInventaire> objets) {
		this.objets = objets;
	}




}
